File: Tatsuki_Yabe/Q1/Go/brute_force.go
package main

func maxArea1(height []int) int {
	max_area := 0
	n := len(height)
	for p1 := range n {
		for p2 := p1 + 1; p2 < n; p2++ {
			length := min(height[p1], height[p2])
			width := p2 - p1
			area := length * width
			max_area = max(max_area, area)
		}
	}

	return max_area
}

File: Tatsuki_Yabe/Q1/Swift/brute_force.swift
class Solution {
    func maxArea1(_ height: [Int]) -> Int {
        var maxarea = 0
        var n = height.count

        for p1 in 0..<n {
            for p2 in p1+1..<n{
                let lenght = min(height[p1], height[p2])
                let width = p2 - p1
                let area = lenght * width
                maxarea = max(maxarea, area)
            }
        }

        return maxarea
    }
}